File: 0226-invert-binary-tree/0226-invert-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {


    public void invertBTree(TreeNode node){

     if(node == null)
      return;

     TreeNode left = node.left;
     node.left = node.right;
     node.right = left;

     invertBTree(node.left);
     invertBTree(node.right);

     return;    
    }

    
    public TreeNode invertTree(TreeNode root) {

     if(root == null)
      return root;

     invertBTree(root);

     return root;
    }
}

File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def invertBTree(self,node: TreeNode | None):
        if not node:
            return None
        
        self.invertBTree(node.left)
        self.invertBTree(node.right)

        temp_node = node.left
        node.left = node.right
        node.right = temp_node

        return

    def invertTree(self, root: TreeNode | None) -> TreeNode | None:
        
        self.invertBTree(root)

        return root